great_expectations/experimental/datasources/data_asset/data_connector/file_path_data_connector.py

```diff
@@ -0,0 +1,303 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Callable, List, Optional
+
+from great_expectations.core.batch_spec import BatchSpec, PathBatchSpec
+from great_expectations.datasource.data_connector.batch_filter import (
+    BatchFilter,
+    build_batch_filter,
+)
+from great_expectations.datasource.data_connector.util import (
+    map_batch_definition_to_data_reference_string_using_regex,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.data_connector import (
```

Comment:
please consider using init.py to simplify these imports; I'd like to see us treat directories more as packages

Reply:
@donaldheppner Dod you mean `__init__.py` or `init.py`?  In any case, I will have to look into how to do it and see if we already have good examples of the style for doing this.  Thank you.

Reply:
the `__init__.py`

Reply:
@donaldheppner This has been done to a significant degree; only `RegexParser` is currently not in `__init__.py` because it was developed as part of ZEP.  At some point, if/when we move it to a more general location, we can update the import as well.  This work also led to the simplification of the V3 `DataConnector`, whereby `MetricsCalculator`, now not causing circular dependencies, is utilized instead of "copy and paste" metric calculation code.  Thanks.